pkg/service/id_local.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/common"
+	"github.com/cilium/cilium/common/types"
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+var (
+	mutex      lock.RWMutex
+	servicesID = map[uint32]*types.L3n4AddrID{}
+	services   = map[string]uint32{}
+	nextID     = common.FirstFreeServiceID
+	maxID      = common.MaxSetOfServiceID
+)
+
+func newServiceID(svc types.L3n4Addr, id uint32) *types.L3n4AddrID {
```

Comment:
Why not have this `L3n4AddrID` constructor in types?

Reply:
> AFAIK the --lb option should only be used in the cilium-agents that are running as LB. Which I agree.

All agents must run with --lb flag. It is targeted for NodePort so all nodes would require this flag anyway. I don't think we have a use case where some nodes run with --lb and and others don't.

Reply:
fixed

Reply:
looked at this again and the function is quite specific to where it is used. I don't think it will ever be reused so I kept it where it is.